src/main/java/com/google/devtools/build/lib/vfs/AbstractFileSystem.java

```diff
@@ -100,18 +99,16 @@ private InputStream createMaybeProfiledInputStream(PathFragment path) throws IOE
 
   @Override
   protected SeekableByteChannel createReadWriteByteChannel(PathFragment path) throws IOException {
-    String name = path.getPathString();
-
     boolean shouldProfile = profiler.isActive() && profiler.isProfiling(ProfilerTask.VFS_OPEN);
 
     long startTime = Profiler.nanoTimeMaybe();
 
     try {
       // Currently, we do not proxy SeekableByteChannel for profiling reads and writes.
-      return Files.newByteChannel(Paths.get(name), READ_WRITE_BYTE_CHANNEL_OPEN_OPTIONS);
+      return Files.newByteChannel(getNioPath(path), READ_WRITE_BYTE_CHANNEL_OPEN_OPTIONS);
     } finally {
       if (shouldProfile) {
-        profiler.logSimpleTask(startTime, ProfilerTask.VFS_OPEN, name);
+        profiler.logSimpleTask(startTime, ProfilerTask.VFS_OPEN, path.toString());
```

Comment:
The writes ultimately end up here: https://github.com/bazelbuild/bazel/blob/424ff69579497768206bea1ae750b4add60262f3/src/main/java/com/google/devtools/build/lib/profiler/JsonTraceFileWriter.java#L227
It seems easier and more consistent to just have this `OutputStreamWriter` use Latin-1. What do you think? I would prefer to defer the changes that only affect "logging" in the broadest sense to a later PR.

Reply:
I like that idea! It's much less error-prone, because in reality most Bazel developers (myself included) will forget to reencode error messages containing filenames some of the time.

Agreed wrt deferring all logging-related issues to a later PR.